Makefile

```diff
@@ -445,13 +445,23 @@ ifeq ($(CI), true)
 	gocov convert $(REPORTS)/$(TEST_SCENARIO)_coverage.out | gocov-xml > $(REPORTS)/$(TEST_SCENARIO)_coverage.xml
 endif
 
-unit-test:
+docker-db:
 	docker ps -q --filter "name=postgres" | xargs -r docker kill && sleep 3
 	docker run -d  --rm --tmpfs /var/lib/postgresql/data --name postgres -e POSTGRESQL_ADMIN_PASSWORD=admin -e POSTGRESQL_MAX_CONNECTIONS=10000 -p 127.0.0.1:5432:5432 \
 	quay.io/edge-infrastructure/postgresql-12-centos7:latest
 	timeout 5m ./hack/wait_for_postgres.sh
-	SKIP_UT_DB=1 $(MAKE) _test TEST_SCENARIO=unit TIMEOUT=30m TEST="$(or $(TEST),$(shell go list ./... | grep -v subsystem))" || (docker kill postgres && /bin/false)
+
+just-unit-test:
+	SKIP_UT_DB=1 $(MAKE) _test TEST_SCENARIO=unit TIMEOUT=30m TEST="$(or $(TEST),$(shell go list ./... | grep -v subsystem))"
+
+ci-unit-test:
+	./hack/setup_db.sh
+	$(MAKE) just-unit-test
+
+kill-db-container:
 	docker kill postgres
+ 
+unit-test: docker-db just-unit-test kill-db-container
```

Comment:
Have you locally tested both `make unit-test` and `skipper make unit-test`?
I think people probably do both.

Reply:
I have always run existing unit tests with skipper. I tested running `make unit-test` locally from the master branch and it fails with a timeout. So, I think running locally needs `skipper` for exisiting `unit-test` target. 

```
~/go/src/github.com/openshift/assisted-service (master) $ make unit-test
docker ps -q --filter "name=postgres" | xargs -r docker kill && sleep 3
docker run -d  --rm --tmpfs /var/lib/postgresql/data --name postgres -e POSTGRESQL_ADMIN_PASSWORD=admin -e POSTGRESQL_MAX_CONNECTIONS=10000 -p 127.0.0.1:5432:5432 \
quay.io/edge-infrastructure/postgresql-12-centos7:latest
58819e63b8465af7ceb6ae4c4274c52f2bc2ef9a5f142c59f369409bb180f602
timeout 5m ./hack/wait_for_postgres.sh
make: *** [Makefile:451: unit-test] Error 124
```

Reply:
@omertuc wasn't this what we were trying to handle in the last PR?

You were running unit tests without skipper right?

Reply:
I don't personally run without skipper, but I didn't want to lock our unit-tests flow to only RPM-based distros (our skipper build container technically being an "RPM-based" distro). 

It was nice when the unit tests flow launched postgres in a container, i.e. assuming the user has podman/docker is better  than assuming the user has a yum package manager

I wasn't aware `make unit-test` already doesn't work. Looking into it it simply has to do with the fact that `hack/wait_for_postgres.sh` silently fails when the host doesn't have a `psql` binary. It should not assume such thing, it should instead run the `psql` binary from inside the container to do this check. It works in skipper because we already [install](https://github.com/openshift/assisted-service/blob/56e04f4305f0514451a4b3ca9b47e70636344dbe/hack/setup_env.sh#L47) an entire `postgresql` package presumably just for this `psql` binary used in wait_for_postgres